pyccel/codegen/printing/fcode.py

```diff
@@ -452,28 +456,112 @@ def _print_Import(self, expr):
         return self._get_statement(code) + '\n'
 
     def _print_PythonPrint(self, expr):
+        end = '\n'
+        sep = ' '
+        code = ''
+        empty_end = FunctionCallArgument(LiteralString(''), 'end')
+        space_end = FunctionCallArgument(LiteralString(' '), 'end')
+        kwargs = [f for f in expr.expr if f.has_keyword]
+        for f in kwargs:
+            if f.keyword == 'sep':
+                sep = str(f.value)
+            elif f.keyword == 'end':
+                end = str(f.value)
+        args_format = []
         args = []
-        n = len(expr.expr)
-        for j, f in enumerate(expr.expr):
+        orig_args = [f for f in expr.expr if not f.has_keyword]
+
+        def formatted_args_to_print(args_format, args, end):
+            if args_format == ['*']:
+                return ', '.join(['print *', *args]) + '\n'
+            args_format = ' A '.join(args_format)
+            new_line = "yes" if end.count('\n') > 0 else "no"
+            end = end.replace('\n', '')
+            args_code = ', " " ,'.join([*args])
+            if end != '':
+                return "write(*, '({},A)',advance=\"{}\") {}, \"{}\"\n".format(args_format, new_line, args_code, end)
+            return "write(*, '({})',advance=\"{}\") {}\n".format(args_format, new_line, args_code)
```

Comment:
this code is broken for multiple line.
test with `print("alloha", 'toto', sep='performed by', end=''\n grr  \n  gr gr \n'')`.

Reply:
@jalalium I have marked this PR as a draft as it seems fixing this is not trivial

Reply:
@rakati thank you, i will try to fix it.
@EmilyBourne thank you!

Reply:
you may find some problems if the end or sep contains special characters, take a look at the  `_print_LiteralString`  method in `fcode.py` to know how to deal with that.

Reply:
don't forget to activate the formated print tests. you will find the test related to that at `tests/pyccel/test_pyccel.py`.
just remove pytest xfail mark for the function `test_print_sp_and_end`.